main.py

```diff
@@ -1,45 +1,42 @@
-import requests
 import json
 from flask import Flask, request
-app = Flask(__name__)
-
-REQUIRED_LABELS = ['ready for review']
+from PullRequest import PullRequest
+from config import REQUIRED_LABELS_ALL, REQUIRED_LABELS_ANY, BANNED_LABELS
 
-# config = {
-#     "all": ["label1", "label2"],
-#     "not": ["label3"],
-#     "any": ["label4", "label5"]
-# }
+app = Flask(__name__)
 
 
-@app.route('/', methods=["POST"])
+@app.route('/', methods=["POST", "GET"])
 def main():
-    pull_request = PullRequest(request.get_json())
-    return create_status_json(any(l['name'] in REQUIRED_LABELS for l in pull_request.labels))
+    #pull_request = PullRequest(request.get_json())
+    pull_request = PullRequest()
+    pull_request.set_issue_url('https://api.github.com/repos/dimagi/commcare-android/issues/1858')
+    return str(create_status_json(pull_request, REQUIRED_LABELS_ANY, REQUIRED_LABELS_ALL, BANNED_LABELS))
 
 
-def create_status_json(has_required_labels):
-    if has_required_labels:
-        description = "Your PR has the necessary labels"
+def create_status_json(pull_request, required_any, required_all, banned):
+    passes_label_requirements = pull_request.validate_labels(required_any, required_all, banned)
+    if passes_label_requirements:
+        description = "PR has the necessary labels"
     else:
-        description = "Your PR requires 1 of the following labels: {}".format(", ".join(REQUIRED_LABELS))
+        description = "PR does not pass the label requirements for this repo --> {}".format(
+            construct_detailed_failure_message(required_any, required_all, banned))
     response_json = {
-        "state": "success" if has_required_labels else "failure",
+        "state": "success" if passes_label_requirements else "failure",
         "target_url": "",
         "description": description,
         "context": "Required-Labels Status Checker"
     }
     return json.dumps(response_json)
 
 
-class PullRequest(object):
-    def __init__(self, event):
-        self.event = event
-
-    @property
-    def label_url(self):
-        return "{}/labels".format(self.event['pull_request']['issue_url'])
+def construct_detailed_failure_message(required_any, required_all, banned):
+    requirement_details = []
+    if required_any != '':
+        requirement_details.append('[1 of the following labels: {}]'.format(', '.join(required_any)))
```

Comment:
is there a reason these error messages have square brackets around them?

Reply:
I was just fiddling with how to make them most readable/consumable when they show up to the user. Feel free to iterate on it, I definitely don't think it's great.